security/pkg/pki/util/keycertbundle.go

```diff
@@ -254,29 +254,29 @@ func (b *KeyCertBundle) UpdateVerifiedKeyCertBundleFromFile(certFile string, pri
 }
 
 // ExtractRootCertExpiryTimestamp returns the unix timestamp when the root becomes expires.
-func (b *KeyCertBundle) ExtractRootCertExpiryTimestamp() (float64, error) {
+func (b *KeyCertBundle) ExtractRootCertExpiryTimestamp() (*time.Time, error) {
 	return extractCertExpiryTimestamp("root cert", b.GetRootCertPem())
 }
 
 // ExtractCACertExpiryTimestamp returns the unix timestamp when the cert chain becomes expires.
-func (b *KeyCertBundle) ExtractCACertExpiryTimestamp() (float64, error) {
+func (b *KeyCertBundle) ExtractCACertExpiryTimestamp() (*time.Time, error) {
 	return extractCertExpiryTimestamp("CA cert", b.GetCertChainPem())
 }
 
 // TimeBeforeCertExpires returns the time duration before the cert gets expired.
 // It returns an error if it failed to extract the cert expiration timestamp.
-// The returned time duration could be a negative value indicating the cert has already been expired.
 func TimeBeforeCertExpires(certBytes []byte, now time.Time) (time.Duration, error) {
 	if len(certBytes) == 0 {
 		return 0, fmt.Errorf("no certificate found")
 	}
 
 	certExpiryTimestamp, err := extractCertExpiryTimestamp("cert", certBytes)
+
 	if err != nil {
 		return 0, fmt.Errorf("failed to extract cert expiration timestamp: %v", err)
 	}
 
-	certExpiry := time.Duration(certExpiryTimestamp-float64(now.Unix())) * time.Second
+	certExpiry := certExpiryTimestamp.Sub(now)
```

Comment:
I think this is more idiomatic because we're not converting back and forth with units. However, it does add nanosecond precision to the measurements.

Reply:
> However, it does add nanosecond precision to the measurements.

Not too familiar with this code, but are there any integration tests or other parts of the code that depend on the precision being seconds as opposed to nanoseconds? I could see this breaking that. I'd confirm this just to be safe.

Reply:
It does end up getting used for rootCertRotation

Reply:
Is there a strong reason for switching from seconds to nanoseconds ?

Reply:
Is there a strong reason for switching from seconds to nanoseconds ?

Reply:
Is there a strong reason for switching from seconds to nanoseconds ?

Reply:
@whitneygriffith No, its just a consequence of not going back and forth with seconds.

Reply:
Using time.Duration seems the right thing to do - the nanosecond should be 0, but that doesn't mean anything.